posydon/unit_tests/utils/test_common_functions.py

```diff
@@ -0,0 +1,2065 @@
+        LBT = totest.LOG10_BURNING_THRESHOLD
+        for sH1 in [2.0*THNA, THNA, 0.5*THNA]:
+            for cH1 in [2.0*TCA, TCA, 0.5*TCA]:
+                for cHe4 in [2.0*TCA, TCA, 0.5*TCA]:
+                    for cC12 in [2.0*TCA, TCA, 0.5*TCA]:
+                        for lgLH in [2.0*LBT, LBT, 0.5*LBT]:
+                            for lgLHe in [2.0*LBT, LBT, 0.5*LBT]:
+                                if sH1>THNA:
+                                    rich = "H-rich"
+                                elif sH1<cH1:
+                                    rich = "accreted_He"
+                                else:
+                                    rich = "stripped_He"
+                                if cH1<=TCA and cHe4<=TCA and cC12<=TCA:
+                                    burn = "Central_C_depletion"
+                                elif cH1<=TCA and cHe4<=TCA and cC12>TCA:
+                                    burn = "Central_He_depleted"
+                                elif cH1>TCA and lgLH>LBT:
+                                    burn = "Core_H_burning"
+                                elif cH1>TCA and lgLH<=LBT:
+                                    burn = "non_burning"
+                                elif lgLHe>LBT:
+                                    burn = "Core_He_burning"
+                                elif lgLH>LBT:
+                                    burn = "Shell_H_burning"
+                                else:
+                                    burn = "non_burning"
+                                assert totest.infer_star_state(surface_h1=sH1,\
+                                       center_h1=cH1, center_he4=cHe4,\
+                                       center_c12=cC12, log_LH=lgLH,\
+                                       log_LHe=lgLHe, log_Lnuc=lgLH+lgLHe) ==\
+                                       rich+"_"+burn
+
+    def test_infer_mass_transfer_case(self, capsys):
+        # missing argument
+        with raises(TypeError, match="missing 3 required positional"+\
+                    " arguments: 'rl_relative_overflow',"+\
+                    " 'lg_mtransfer_rate', and 'donor_state'"):
+            totest.infer_mass_transfer_case()
+        # examples: no RLO
+        assert totest.infer_mass_transfer_case(None, None, None) ==\
+               totest.MT_CASE_NO_RLO
+        assert totest.infer_mass_transfer_case(None, 1.0, "test") ==\
+               totest.MT_CASE_NO_RLO
+        assert totest.infer_mass_transfer_case(1.0, None, "test") ==\
+               totest.MT_CASE_NO_RLO
+        RROT = totest.RL_RELATIVE_OVERFLOW_THRESHOLD
+        LMRT = totest.LG_MTRANSFER_RATE_THRESHOLD
+        assert totest.infer_mass_transfer_case(min(-1.0, RROT), LMRT, "test")\
+               == totest.MT_CASE_NO_RLO
+        assert totest.infer_mass_transfer_case(min(-1.0, RROT), LMRT, "test",\
+               verbose=True) == totest.MT_CASE_NO_RLO
+        assert "checking rl_relative_overflow / lg_mtransfer_rate" in\
```

Comment:
It shouldn't hurt to verify the output you get, when you request more output.

Reply:
We can leave it here, but it should not be standard, since debug output can change often.
Additionally, it's not crucial to the code's functioning. I think we discussed that it wasn't necessary to test.

Reply:
Here we should differentiate between debugging output and verbose output: the first one is opt to many changes for developers to get more information to pin-point a bug down. I prefer to turn them into a comment when currently not needed. The latter one is additional out for users, which should be checked that users can rely on getting that output when requested. Usually, all the output giving internal stuff as indexes are more of debugging nature, while outputs stating otherwise hidden physical values I'd count as user output.

Reply:
I think about this differently, but it's fine to keep the test here.